src/main/java/mahaveer/Storage.java

```diff
@@ -0,0 +1,226 @@
+package mahaveer;
+
+import mahaveer.task.Task;
+import mahaveer.task.Deadline;
+import mahaveer.task.Event;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The Storage class is responsible for loading and saving tasks to a file.
+ * It provides methods to add, delete, mark, and unmark tasks.
+ */
+public class Storage {
+    private static final String FILE_PATH = "./data/tasks.txt";
+
+    /**
+     * Constructs a new Storage instance.
+     * <p>
+     * This constructor ensures the storage file and its directory exist.
+     * If the file is created for the first time, it prints a confirmation message.
+     * </p>
+     */
+    public Storage() {
+        File file = new File(FILE_PATH);
+        try {
+            file.getParentFile().mkdirs(); // Ensure the directory exists
+            if (file.createNewFile()) {
+                System.out.println("Storage file created: " + FILE_PATH);
+            }
+        } catch (IOException e) {
+            System.out.println("Error creating storage file: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads tasks from the storage file.
+     *
+     * @return a list of tasks loaded from the file; returns an empty list if the file does not exist or an error occurs.
+     */
+    public List<Task> loadTasks() {
+        List<Task> taskList = new ArrayList<>();
+        File file = new File(FILE_PATH);
+        if (!file.exists()) {
+            return taskList;
+        }
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(FILE_PATH))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task task = parseTask(line);
+                if (task != null) {
+                    taskList.add(task);
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Error loading tasks: " + e.getMessage());
+        }
+        return taskList;
+    }
+
+    /**
+     * Parses a line from the storage file into a Task object.
+     *
+     * @param line a line from the file representing a task in a specific format.
+     * @return a Task object if parsing is successful; otherwise, returns null.
+     */
+    private Task parseTask(String line) {
+        String[] parts = line.split(", ");
+        if (parts.length < 3) return null;
```

Comment:
conditional statements should be put inside braces, even if they are single line ones.

Reply:
applies to all such instances in the codebase